Solutions/STM32F4DISCOVERY/scatterfile_gcc_missing_symbols.xml

```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <ScatterFile xmlns="http://schemas.microsoft.com/netmf/ScatterfileSchema.xsd">
+  <!-- 
+    - using newlib-nano adds libnosys.a, which calls sbrk, which requires the 'end' symbol
+    - this is pulled in by a _printf imported in function _Crypto_StepRSAOperation @ Crypto.lib(rsa_zenith.obj) 
+  -->
+  <Provide Name="end" />

```

Comment:
Thanks for the explanation.

I am a little bit surprised that the printf is referenced from crypto function, because if I understand it correctly, the solution should not use it (I definitely don't have crypto.lib anywhere, and the one in the build output is copy of crypto_stub.lib; crypto dotNetMF.proj is somewhat weird).

I'll have a closer look at it, maybe I am wrong...

Nonetheless, the proprietary crypto code should be replaced with an appropriate alternative (along with OpenSSL). Perhaps it's time to check out mbed TLS...